echo_server.py

```diff
@@ -0,0 +1,67 @@
+from __future__ import unicode_literals
+import socket
+
+
+def response_ok(uri):
+    response = []
+    response.append("HTTP/1.1 200 OK")
+    response.append("Content-Type = text/html; charset=utf-8")
+    response.append("")
+    response.append(uri)
+
+    response = "\r\n".join(response).encode("utf-8")
+    return response
+
+
+def response_error(error, error_msg):
+    response = []
+    response.append("HTTP/1.1 {} {}".format(error, error_msg))
+    response.append("")
+    response.append("{} {}".format(error, error_msg))
+    response = "\r\n".join(response).encode("utf-8")
+
+    return response
+
+
+def parse_request(request):
+    first_line = request.splitlines()[0]
+    first_line = first_line.split(" ")
+
+    if first_line[0] == "GET":
+        if first_line[2] == "HTTP/1.1":
+            return response_ok(first_line[1])
+        else:
+            return response_error(505, "Protocol must be HTTP/1.1")
+    else:
+        return response_error(405, "Method must be GET")
+
+
+if __name__ == "__main__":
+    server_socket = socket.socket(
+        socket.AF_INET,
+        socket.SOCK_STREAM,
+        socket.IPPROTO_IP)
+
+    buffer_size = 32
+
+    server_socket.bind(('127.0.0.1', 50000))
+    server_socket.listen(1)
+
+    try:
+        while True:
+            request = ''
+            conn, addr = server_socket.accept()
+
+            request_part = conn.recv(buffer_size).decode('utf-8')
+            request += request_part
+            while len(request_part) >= buffer_size:
```

Comment:
request_part will never ever be greater than buffer_size.  It would break the specification for sockets if it were.